sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1450,7 +1558,7 @@ def __init__(
         min_samples_leaf=20,
         l2_regularization=0.0,
         max_bins=255,
-        categorical_features=None,
+        categorical_features="by_dtype",
```

Comment:
The first line in the docstring for `categorical_features` needs to be updated accordingly. It's still written `default=None`.